### DISCUSSION TITLE
Flashing firmware to a self-built Open Deck L 3.1.0

### DISCUSSION THREAD

**Discussion by sam-schorb** [shanteacontrols/OpenDeck#198] [General]
I decided to use the CAD files you so kindly open-sourced to build my own Open Deck L, from the v3.1.0 KiCAD repository files. I'd be happy to donate to your Patreon/Paypal or something if you have one?

The hardware side is going well but I've only ever flashed firmware with simple web interfaces before.

What do I need to do to get the latest firmware onto the board? I've got some idea from the other posts here but not sure if they would work in my situation. If you could provide instructions on how I connect to the board and what I need to do with which files it would be greatly appreciated. Will the USB power work if there is no previous firmware on the RP2040?

I have Windows 10 desktop or an M1 Macbook Air 2020 to work on. Which would you recommend using for this process? I also have an ST-Link I've never used, might this be useful or do I need a different programmer?

Thank you for this great project.

**Comment by paradajz** [comment]
The only thing you need to do is to copy the opendeck31.uf2 file to the RPI disk drive which will show the first time you plug the board in.

**Reply by sam-schorb** [reply]
Thank you very much for this very quick reply. Just to be sure, you mean I should do this?:

1. Make a USB connection between the USB-C connector on the Open Deck.
2. The RPi disk drive will come up as an external drive on the computer.
3. Copy opendeck31.uf2 file into this drive.
4. Turn the board off and on again.

And after I do this the Open Deck will function correctly and it will be possible to configure it with the web configurator interface?
There is no need for any other hardware such as ST-link?
Is there any way I can make a financial donation to you to thank you?

Sorry for the newb questions.

**Reply by paradajz** [reply]
> Thank you very much for this very quick reply. Just to be sure, you mean I should do this?:
> 
> 1. Make a USB connection between the USB-C connector on the Open Deck.
> 2. The RPi disk drive will come up as an external drive on the computer.

Yes.

> 3. Copy opendeck31.uf2 file into this drive.

Yes.

> 4. Turn the board off and on again.

No. After you copy the uf2 file to the drive the board will reboot and perform initial setup which takes few seconds. After that a new OpenDeck device will be present on your PC.
 
> And after I do this the Open Deck will function correctly and it will be possible to configure it with the web configurator interface?

Yes.

> There is no need for any other hardware such as ST-link?

No. You are confusing STM32 MCUs with the one I'm using on the new OpenDeck boards which is Raspberry Pi RP2040. RP2040 has factory bootloader which allows you to flash the binary without usage of an external programmer.

> Is there any way I can make a financial donation to you to thank you?

Not needed. The only form of "donation" I accept is purchasing of official boards through Tindie. ;)

**Comment by azgael** [comment]
Maybe one missing step is that: to bring up the RP2040 boot-loader you must short the jumper next to the usb connector before you plug the USB cable to the pc, them after you have a kind of flash drive on the pc open you should remove the short, then you can transfer the *.uf2 to the flash drive and you are done!
:)
azgael

**Reply by paradajz** [reply]
True, but if the chip is blank (as it comes from the factory) the bootloader will automatically be loaded.
